osc/connection.py

```diff
@@ -342,6 +342,14 @@ def http_request(method: str, url: str, headers=None, data=None, file=None):
         http.client.print(40 * '-')
         http.client.print(method, url)
 
+    if method != "GET" and os.environ['OSC_HTTP_MANUAL_APPROVE']:
+        print("osc is going to send "+ method +" request to " + urlopen_url)
+        if data:
+            print(data.decode("utf-8"))
+        y = input("press 'y' to continue or any other key to cancel")
+        if y != 'y' and y != 'Y':
+            raise Exception("Manual abort")
+
```

Comment:
We should use functions and exceptions osc already offers:
(please excuse the wrong indendation, I've copied the code from a different place and slightly modified it to fit the scenario)

```
               from .output import get_user_input

               reply = get_user_input(
                    f"Do you want to  make '{method}' request to '{urlopen_url}'?",
                    answers={"y": "yes", "n": "no"},
                    default_answer="n",
                )
                if reply == "n":
                    raise oscerr.UserAbort()
```